packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -392,40 +392,54 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
     private async summarizeInternal(fullTree: boolean, trackState: boolean): Promise<ISummarizeInternalResult> {
         await this.realize();
 
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        const summarizeResult = await this.channel!.summarize(fullTree, trackState);
-        let pathPartsForChildren: string[] | undefined;
+        // Create attributes to be added to the summary.
+        const { pkg, isRootDataStore } = await this.getInitialSnapshotDetails();
 
-        if (!this.disableIsolatedChannels) {
-            // Wrap dds summaries in .channels subtree.
-            wrapSummaryInChannelsTree(summarizeResult);
-            pathPartsForChildren = [channelsTreeName];
+        let summarizeResult;
+        try {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Instead of this can you assert that `this.channel` is not undefined similar to how other places do it.